mne/time_frequency/psd.py

```diff
@@ -11,17 +11,26 @@
 from .multitaper import _psd_multitaper
 
 
-def _psd_func(epoch, noverlap, nfft, fs, freq_mask, func):
+def _psd_func(epoch, noverlap, n_per_seg, nfft, fs, freq_mask, func):
     """Aux function."""
-    return func(epoch, fs=fs, nperseg=nfft, noverlap=noverlap,
+    return func(epoch, fs=fs, nperseg=n_per_seg, noverlap=noverlap,
                 nfft=nfft, window='hann')[2][..., freq_mask, :]
 
 
-def _check_nfft(n, n_fft, n_overlap):
-    """Helper to make sure n_fft and n_overlap make sense."""
-    n_fft = n if n_fft > n else n_fft
-    n_overlap = n_fft - 1 if n_overlap >= n_fft else n_overlap
-    return n_fft, n_overlap
+def _check_nfft(n, n_fft, n_per_seg, n_overlap):
+    """Helper to make sure n_fft, n_per_seg and n_overlap make sense."""
+    if n_per_seg is None and n_fft > n:
+        msg = ('If n_per_seg is None n_fft is not allowed to be > n_times. If '
+               'you want zero-padding, you have to set n_per_seg to relevant '
+               'length. Got n_fft of {} while signal length is {}.')
+        raise ValueError(msg.format(n_fft, n))
+    n_per_seg = n_fft if n_per_seg is None or n_per_seg > n_fft else n_per_seg
+    n_per_seg = n if n_per_seg > n else n_per_seg
+    if n_overlap >= n_per_seg:
+        msg = ('n_overlap cannot be greater than n_per_seg (or n_fft). Got '
+               'n_overlap of {} while n_per_seg is {}.')
```

Comment:
FYI for future reference by convention we 1. don't split these across multiple vars/commands and 2. use/prefer `%` (even though there is talk of deprecation)